apps/api/src/app/portfolio/calculator/twr/portfolio-calculator.ts

```diff
@@ -754,126 +121,19 @@ export class PortfolioCalculator {
-    const symbols: { [symbol: string]: TransactionPointSymbol } = {};
-
-    let lastDate: string = null;
-    let lastTransactionPoint: TransactionPoint = null;
-
-    for (const {
-      fee,
-      date,
-      quantity,
-      SymbolProfile,
-      tags,
-      type,
-      unitPrice
-    } of this.orders) {
-      let currentTransactionPointItem: TransactionPointSymbol;
-      const oldAccumulatedSymbol = symbols[SymbolProfile.symbol];
-
-      const factor = getFactor(type);
-
-      if (oldAccumulatedSymbol) {
-        let investment = oldAccumulatedSymbol.investment;
-
-        const newQuantity = quantity
-          .mul(factor)
-          .plus(oldAccumulatedSymbol.quantity);
-
-        if (type === 'BUY') {
-          investment = oldAccumulatedSymbol.investment.plus(
-            quantity.mul(unitPrice)
-          );
-        } else if (type === 'SELL') {
-          investment = oldAccumulatedSymbol.investment.minus(
-            quantity.mul(oldAccumulatedSymbol.averagePrice)
-          );
-        }
-
-        currentTransactionPointItem = {
-          investment,
-          tags,
-          averagePrice: newQuantity.gt(0)
-            ? investment.div(newQuantity)
-            : new Big(0),
-          currency: SymbolProfile.currency,
-          dataSource: SymbolProfile.dataSource,
-          dividend: new Big(0),
-          fee: fee.plus(oldAccumulatedSymbol.fee),
-          firstBuyDate: oldAccumulatedSymbol.firstBuyDate,
-          quantity: newQuantity,
-          symbol: SymbolProfile.symbol,
-          transactionCount: oldAccumulatedSymbol.transactionCount + 1
-        };
-      } else {
-        currentTransactionPointItem = {
-          fee,
-          tags,
-          averagePrice: unitPrice,
-          currency: SymbolProfile.currency,
-          dataSource: SymbolProfile.dataSource,
-          dividend: new Big(0),
-          firstBuyDate: date,
-          investment: unitPrice.mul(quantity).mul(factor),
-          quantity: quantity.mul(factor),
-          symbol: SymbolProfile.symbol,
-          transactionCount: 1
-        };
-      }
-
-      symbols[SymbolProfile.symbol] = currentTransactionPointItem;
-
-      const items = lastTransactionPoint?.items ?? [];
-
-      const newItems = items.filter(({ symbol }) => {
-        return symbol !== SymbolProfile.symbol;
-      });
-
-      newItems.push(currentTransactionPointItem);
-
-      newItems.sort((a, b) => {
-        return a.symbol?.localeCompare(b.symbol);
-      });
-
-      if (lastDate !== date || lastTransactionPoint === null) {
-        lastTransactionPoint = {
-          date,
-          items: newItems
-        };
-
-        this.transactionPoints.push(lastTransactionPoint);
-      } else {
-        lastTransactionPoint.items = newItems;
-      }
-
-      lastDate = date;
-    }
-  }
-
-  private getSymbolMetrics({
+  protected getSymbolMetrics({
     dataSource,
     end,
     exchangeRates,
-    isChartMode = false,
+    isChartMode,
```

Comment:
Why has this changed?

Reply:
`A parameter initializer is only allowed in a function or constructor implementation.ts (2371)`

what I need to make sure is that when using this method `calculateOverallPerformance` on TWR this default value is set